### PR TITLE
Log backoff responses (HTTP 429 Too Many Requests) as a warning to th…

### PR DESCRIPTION
**Pull request by bretthoerner**
…e `lockdown` logger instead of an error.

### CODE REVIEW

**Comment by adrian-baker** [comment]
> (https://github.com/getsentry/sentry-java/issues/559#issuecomment-367736858 ): 
> 
> I spammed Sentry and this was the (new) result:
> 
> 10:19:08.665 [sentry-pool-1-thread-2] WARN  io.sentry.SentryClient.lockdown - Too many requests to Sentry, backing off by temporarily locking down the connection. See: https://docs.sentry.io/learn/quotas/
> Dropping an Event due to lockdown: Event{level=ERROR, message='foo', logger='null'}
> 10:19:08.696 [sentry-pool-1-thread-3] WARN  io.sentry.SentryClient.lockdown - Dropping an Event due to lockdown: Event{level=ERROR, message='foo', logger='null'}
> 10:19:08.746 [sentry-pool-1-thread-1] WARN  io.sentry.SentryClient.lockdown - Dropping an Event due to lockdown: Event{level=ERROR, message='foo', logger='null'}
> 10:19:08.796 [sentry-pool-1-thread-2] WARN  io.sentry.SentryClient.lockdown - Dropping an Event due to lockdown: Event{level=ERROR, message='foo', logger='null'}

I don't think the high volume per event warnings ("Dropping an Event") should be logged by at WARN, but instead DEBUG. The initial lockdown message is sufficient, subsquently logging individual warnings for a high volume of events (otherwise you wouldn't have been throttled) isn't a good default. I believe this was the previous behavior in the com.getraven version.

If you did keep logging them at WARN, they would need to be on a separate logger to the lockdown message so they could be disabled.

File: sentry/src/main/java/io/sentry/connection/AsyncConnection.java [hunk 1]
```diff
@@ -172,7 +172,7 @@ public void run() {
                 // The current thread is managed by sentry
                 actualConnection.send(event);
             } catch (LockedDownException e) {
-                lockdownLogger.warn("The connection to Sentry is currently locked down.", e);
+                lockdownLogger.warn(e.getMessage());
```
Comment (adrian-baker) [inline sentry/src/main/java/io/sentry/connection/AsyncConnection.java]: I would always suggest logging the full exception at DEBUG level for troubleshooting.

**Comment by adrian-baker** [comment]
>I don't think the high volume per event warnings ("Dropping an Event") should be logged by at WARN, but instead DEBUG.

This would also mean you could retain the stack trace in these logs.

File: sentry/src/main/java/io/sentry/connection/AsyncConnection.java (hunk 1 above)
Comment (bretthoerner) [reply sentry/src/main/java/io/sentry/connection/AsyncConnection.java]: Good call.

**Comment by bretthoerner** [comment]
I did a bit of an overhaul. I also realized the server response of `Retry-After` must have changed and we were failing to parse it. :( That will help with backoffs.

Anyway, dropped events are currently at debug. Lockdown initiation is at warning (with no stacktrace). They occur from different classes (with a `.lockdown` suffix) so they can be adjusted independently.

**Pull request merged**